`safe_int_cast` breaks builds on i386

The check at https://github.com/Komet/MediaElch/blob/master/src/utils/Meta.h#L234 breaks builds on i386:

```
/<<PKGBUILDDIR>>/src/utils/Meta.h: In instantiation of ‘constexpr To safe_int_cast(From) [with To = int; From = int]’:
/<<PKGBUILDDIR>>/src/utils/Meta.h:208:30:   required from here
/<<PKGBUILDDIR>>/src/utils/Meta.h:194:44: error: static assertion failed: Unnecessary int cast: Both types are the same
  194 |     static_assert(!std::is_same<From, To>::value, "Unnecessary int cast: Both types are the same");
      |                                            ^~~~~
/<<PKGBUILDDIR>>/src/utils/Meta.h:194:44: note: ‘!(bool)std::integral_constant<bool, true>::value’ evaluates to false
```

A full build log is at: https://salsa.debian.org/rvandegrift/mediaelch/-/jobs/3681235/raw

**To Reproduce**
Steps to reproduce the behavior:
1. build on i386

**Expected behavior**
Casts that become redundant due to machine word size should be okay.

**MediaElch Version:**
 - [ ] 2.8.19-dev (nightly)
 - [X] 2.8.18 stable
<!-- older stable versions are not supported; please update -->

**Operating System:**
 - [ ] Windows
 - [ ] macOS
 - [X] Linux (distribution: debian sid)

--- a/src/utils/Meta.h
+++ b/src/utils/Meta.h
@@ -231,9 +231,14 @@ struct int_conversion<Signed, Unsigned, From, To>
 template<typename To, typename From>
 constexpr To safe_int_cast(From from)
 {
+    // These checks are sanity checks used while upgrading from Qt5 to Qt6.
+    // They don't work properly on 32bit systems, so don't run them there.
+    // See https://github.com/Komet/MediaElch/issues/1511
+#ifdef __x86_64__
     static_assert(!std::is_same<From, To>::value, "Unnecessary int cast: Both types are the same");
     static_assert(std::is_integral<From>::value, "safe_int_cast failed: From type is not integral");
     static_assert(std::is_integral<To>::value, "safe_int_cast failed: To type is not integral");
+#endif
     return detail::int_conversion<std::is_signed<From>::value, std::is_signed<To>::value, From, To>::convert(from);
 }
 

